File: app/src/main/java/com/tkdev/weatherapp/dialogs/ShareDialog.java
package com.tkdev.weatherapp.dialogs;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.CheckBox;

import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import com.tkdev.weatherapp.R;

import java.util.ArrayList;
import java.util.Objects;

public class ShareDialog extends DialogFragment {

    public interface ShareDialogListener {
        void onShareDialogPositiveClick(ArrayList<Boolean> booleanList);

        void onShareDialogNegativeClick(String text);
    }

    ShareDialogListener listener;


    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        try {
            listener = (ShareDialogListener) context;
        } catch (ClassCastException e) {
            throw new ClassCastException(getActivity().toString()
                    + " must implement ShareDialogListener");
        }
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(Objects.requireNonNull(getActivity()));
        LayoutInflater inflater = requireActivity().getLayoutInflater();
        View dialogView = inflater.inflate(R.layout.fragment_share, null);
        final CheckBox currentCb = dialogView.findViewById(R.id.share_current_temp);
        final CheckBox weatherCb = dialogView.findViewById(R.id.share_weather);
        final CheckBox lastUpdateCb = dialogView.findViewById(R.id.share_last_update);
        ArrayList<Boolean> share = new ArrayList<>();
        builder.setView(dialogView)
                .setPositiveButton(R.string.apply, (dialog, id) -> {
                    share.add(currentCb.isChecked());
                    share.add(weatherCb.isChecked());
                    share.add(lastUpdateCb.isChecked());
                    listener.onShareDialogPositiveClick(share);
                })
//                .setNegativeButton(R.string.cancel, (dialog, which) -> {
//                    setCancelable(true);
//                    listener.onDialogNegativeClick(getString(R.string.search_dialog_onCancel));
//                })
                ;


        return builder.create();
    }
}

File: app/src/main/java/com/tkdev/weatherapp/common/dialogs/ShareDialog.kt
package com.tkdev.weatherapp.common.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.widget.CheckBox
import androidx.fragment.app.DialogFragment
import com.tkdev.weatherapp.R
import kotlinx.android.synthetic.main.fragment_share.view.*

class ShareDialog: DialogFragment() {

    lateinit var listener: ShareDialogListener

    interface ShareDialogListener {
        fun onSharePositiveClick(list: ArrayList<Boolean>)
        fun onShareNegativeClick(message: String)
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        try {
            listener = context as ShareDialogListener
        } catch (e: ClassCastException) {
            throw ClassCastException((context.toString() +
                    " must implement ShareDialogListener"))
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {


        return activity?.let {
            val builder: AlertDialog.Builder? = AlertDialog.Builder(it)
            val inflater = requireActivity().layoutInflater
            val dialogLayout = inflater.inflate(R.layout.fragment_share, null)
            val currentCheckBox: CheckBox = dialogLayout.share_current_temp
            val weatherCheckBox: CheckBox = dialogLayout.share_weather
            val lastUpdateCheckBox: CheckBox = dialogLayout.share_last_update
            val share = ArrayList<Boolean>()
            builder?.setView(dialogLayout)

            builder?.setPositiveButton(R.string.apply)
            { _, _ ->
                share.add(currentCheckBox.isChecked)
                share.add(weatherCheckBox.isChecked)
                share.add(lastUpdateCheckBox.isChecked)
                listener.onSharePositiveClick(share)
            }
//            builder?.setNegativeButton(R.string.cancel)
//            { _, _ ->
//                listener.onSearchNegativeClick(getString(R.string.search_dialog_onCancel))
//            }

            builder?.create()
        } ?: throw IllegalStateException("Activity cannot be null")
    }
}